sources/react/template-all-icons.js

```diff
@@ -1,52 +1,69 @@
-const toString = array => array.reduce( ( acc, item ) => ( acc = acc + "'" + item + "', " ), '' );
-
-const prepareAllIcons = ( {
-	components,
-	iconsThatNeedOffset,
-	iconsThatNeedOffsetX,
-	iconsThatNeedOffsetY,
-} ) => `
+const toString = array =>
+        array.reduce((acc, item) => (acc = acc + "'" + item + "', "), '');
+
+const prepareAllIcons = ({
+        components,
+        iconsThatNeedOffset,
+        iconsThatNeedOffsetX,
+        iconsThatNeedOffsetY,
+}) => `
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
 
-export default function( { icon: iconProp, className, onClick, size = 24, ...otherProps } ) {
+export default class Gridicon extends PureComponent {
 
-	const iconsThatNeedOffset = [ ${ toString( iconsThatNeedOffset ) } ];
+	static defaultProps = {
+		size: 24
+	};
 
-	const iconsThatNeedOffsetX = [ ${ toString( iconsThatNeedOffsetX ) } ];
+	static propTypes = {
+		icon: PropTypes.string.isRequired,
+		size: PropTypes.number,
+		onClick: PropTypes.func,
+		className: PropTypes.string
+	};
 
-	const iconsThatNeedOffsetY = [ ${ toString( iconsThatNeedOffsetY ) } ];
+	render() {
 
-	const doesItNeedOffset = ( name, icons ) => icons.indexOf( name ) >= 0;
-	const isModulo18 = size => size % 18 === 0;
+		const { size, onClick, icon: iconProp, className, ...otherProps } = this.props;
 
-	const icon = 'gridicons-' + iconProp;
-	const needsOffset = doesItNeedOffset( icon, iconsThatNeedOffset ) && isModulo18( size );
-	const needsOffsetX = doesItNeedOffset( icon, iconsThatNeedOffsetX ) && isModulo18( size );
-	const needsOffsetY = doesItNeedOffset( icon, iconsThatNeedOffsetY ) && isModulo18( size );
+		const iconsThatNeedOffset = [ ${toString(iconsThatNeedOffset)} ];
```

Comment:
might be better to have these as a static property of the class. That way you're not recreating the array on every render.

Reply:
Done at 9af3e7d8ada9e7877e8baaaecf4ebf8e7554b40b